aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -40,12 +31,13 @@ namespace dist {
       );
     }
   };
+
 } // at::native::dist
 
 Tensor _s_poisson_cuda(const Tensor& lambda, Generator* gen) {
   Tensor ret = lambda.type().tensor(lambda.sizes());
   auto lambda_ = lambda.toType(ScalarType::Float);
-  dispatch_all<void, dist::PoissonOpCUDA>(ret.type(), "poisson", ret, lambda_, dist::get_states(gen));
+  dispatch_floating_types<void, dist::PoissonOpCUDA>(ret.type(), "poisson", ret, lambda_, dist::get_states(gen));
```

Comment:
if you are having trouble with half, try `DISPATCH_ALL_FLOATING_TYPES`; `dispatch_all` uses `Half` and `DISPATCH_ALL_FLOATING_TYPES` uses `half`; we should resolve this, probably always using `Half` (and providing __device__ conversion functions?).  CC @colesbury

Reply:
Thanks, that works!

Reply:
Ideally we would always use `half` for device functions and `at::Half` for host functions.